author/topics/automation/json_to_text.adoc

```diff
@@ -0,0 +1,586 @@
+
+The `json2text` block lets you generate document elements directly
+from a separate JSON file into the document. [added in https://github.com/metanorma/metanorma-standoc/releases/tag/v1.5.3]
+
+== Expressions
+
+`json2text` supports all https://shopify.github.io/liquid/basics/introduction/[Liquid syntax expressions], including:
+
+* variables, variable assignment
+* flow control (if/case)
+* filters
+* loops
+
+See https://shopify.github.io/liquid/basics/introduction/[here] for the full description of Liquid tags and expressions.
+
+
+
+== Syntax
+
+[[defining_syntax]]
+=== Defining the block
+
+A `json2text` block is created with the following syntax.
+
+Block opening and closing is demarcated by an open block syntax (`--`)
+or the `[source]` block syntax (`----` or more `-`).
+
+[source,adoc]
+--
+[json2text,{JSON file path},{self-defined context name}]
+----
+this is content within the json2text block!
+----
+--
+
+Where:
+
+* content within the block is called the "`template`";
+
+* `{JSON file path}` is the location of the JSON file that contains data to be loaded. Location of the JSON file is computed relative to the source directory that `[json2text]` is used (e.g., if `[json2text,data.json,data]` is invoked in an `.adoc` file located at `/foo/bar/doc.adoc`, the data file is expected to be found at `/foo/bar/data.json`);
+
+* `{self-defined context name}` is the name where the JSON data read from the JSON file can be accessed with.
+
+=== Interpolation
+
+`json2text` accepts string interpolation of the following forms:
+
+. `{variable}`: as in AsciiDoc syntax;
+. `{{ variable }}`, `{% if/else/for/case %}`: basic Liquid tags and expressions are supported.
+
+The value within the curly braces will be interpolated by `json2text`.
+
+Where:
+
+* In `{variable}`(`{{variable}}`), `variable` is the name of the variable or AsciiDoc attribute.
+* The location of `{variable}`(`{{variable}}`) in text will be replaced with the value of `variable`.
+* Evaluation order will be first from the defined context, then of the Metanorma AsciiDoc document.
+
+
+=== Accessing object values
+
+Object values are accessed via the `.` (dot) separator.
+
+EXAMPLE:
+--
+Given:
+
+strings.json
+[source,json]
+----
+{
+  "foo:" "bar",
+  "dead": "beef"
+}
+----
+
+And the block:
+[source,asciidoc]
+------
+[json2text,strings.json,data]
+----
+I'm heading to the {{data.foo}} for {{data.dead}}.
+----
+------
+
+The file path is `strings.json`, and context name is `data`.
+`{{data.foo}}` evaluates to the value of the key `foo` in `data`.
+
+Will render as:
+[source,asciidoc]
+----
+I'm heading to the bar for beef.
+----
+
+--
+
+
+=== Accessing arrays
+
+==== Length
+
+The length of an array can be obtained by `{{arrayname.size}}`.
+
+EXAMPLE:
+--
+Given:
+
+strings.json
+[source,json]
+----
+---
+[ "lorem", "ipsum", "dolor" ]
+----
+
+And the block:
+[source,asciidoc]
+------
+[json2text,strings.json,data]
+----
+The length of the JSON array is {{data.size}}.
```

Comment:
Is the JSON length really called "size"?

Reply:
data is a `Hash`, not JSON, `size` is a ruby method that works for both Array and Hash